src/wfs/src/test/java/org/geoserver/wfs/response/CSVOutputFormatTest.java

```diff
@@ -468,4 +473,143 @@ public void testIAULayer() throws Exception {
+        gs.save(wfsInfo);
+        CSVOutputFormat format = new CSVOutputFormat(gs);
+        format.write(fct, bos, op);
+
+        // read the response back with a parser that can handle escaping, newlines and what not
+        List<String[]> lines = readLines(bos.toString(), ',');
+
+        // we should have one header line and then all the features in that feature type
+        assertEquals(fs.getCount(Query.ALL) + 1, lines.size());
+
+        for (String[] line : lines) {
+            // check each line has the expected number of elements
+            assertEquals(fs.getSchema().getAttributeCount() + 1, line.length);
+        }
+
+        assertEquals(
+                TemporalUtils.serializeDateTime(
+                        (Date) f1.getAttribute("dtg"), "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"),
+                lines.get(1)[3]);
+
+        ByteArrayOutputStream bos1 = new ByteArrayOutputStream();
+        GeoServer gs1 = getGeoServer();
+        WFSInfo wfsInfo1 = gs1.getService(WFSInfo.class);
+        wfsInfo1.setDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS");
+        gs1.save(wfsInfo1);
+        CSVOutputFormat format1 = new CSVOutputFormat(gs1);
+        format1.write(fct, bos1, op);
+
+        // read the response back with a parser that can handle escaping, newlines and what not
+        List<String[]> lines1 = readLines(bos1.toString(), ',');
+
+        // we should have one header line and then all the features in that feature type
+        assertEquals(fs.getCount(Query.ALL) + 1, lines1.size());
+
+        for (String[] line : lines1) {
+            // check each line has the expected number of elements
+            assertEquals(fs.getSchema().getAttributeCount() + 1, line.length);
+        }
+
+        assertEquals(
+                TemporalUtils.serializeDateTime(
+                        (Date) f1.getAttribute("dtg"), "yyyy-MM-dd'T'HH:mm:ss.SSS"),
+                lines1.get(1)[3]);
+
+        ByteArrayOutputStream bos2 = new ByteArrayOutputStream();
+        GeoServer gs2 = getGeoServer();
+        WFSInfo wfsInfo2 = gs2.getService(WFSInfo.class);
+        wfsInfo2.setDateFormat("EEE, MMM d, ''yy");
+        gs2.save(wfsInfo2);
+        CSVOutputFormat format2 = new CSVOutputFormat(gs2);
+        format2.write(fct, bos2, op);
+
+        // read the response back with a parser that can handle escaping, newlines and what not
+        List<String[]> lines2 = readLines(bos2.toString(), ',');
+
+        // we should have one header line and then all the features in that feature type
+        assertEquals(fs.getCount(Query.ALL) + 1, lines2.size());
+
+        for (String[] line : lines2) {
+            // check each line has the expected number of elements
+            assertEquals(fs.getSchema().getAttributeCount() + 1, line.length);
+        }
+
+        assertEquals(
+                TemporalUtils.serializeDateTime((Date) f1.getAttribute("dtg"), "EEE, MMM d, ''yy"),
+                lines2.get(1)[3]);
```

Comment:
Most of this code is a copy/paste from the section above. In a situation like this, factor out a helper method and pass the differences as parameters instead (in this case, apparently the date format and the expected result).